### PR TITLE
♻️ Add module / CJS and types to purifier npm package

### PR DESCRIPTION
**Pull request by fstanis**

### CODE REVIEW

File: src/purifier/package.json [hunk 1]
```diff
@@ -1,15 +1,23 @@
 {
   "name": "@ampproject/purifier",
-  "version": "1.0.0",
+  "version": "1.0.1",
   "description": "AMP-specific sanitization library",
   "author": "The AMP HTML Authors",
   "license": "Apache-2.0",
-  "main": "dist/purifier.js",
-  "files": ["dist/*"],
+  "main": "dist/purifier.mjs",
```
Comment (choumx) [inline src/purifier/package.json]: Is only having an "mjs" output typical?

Comment (samouri) [reply src/purifier/package.json]: My impression was that `main: ` should be CJS and `module: ` is normally mjs.

Comment (fstanis) [reply src/purifier/package.json]: Nice, didn't know this. Added CJS output now and used `main` and `module` in package.json.

Comment (choumx) [reply src/purifier/package.json]: Looks like this was a bridge until Node.js supported ES6 modules: https://nodejs.org/api/esm.html#esm_dual_commonjs_es_module_packages

Comment (fstanis) [reply src/purifier/package.json]: Reading a bit into this, it seems the `module` + `main` approach @samouri suggested is still best, since Node.js will just ignore `module` and use `main`. The solution from nodejs.org to have a wrapper instead seems only useful to allow ESM support in Node (which is still experimental in LTS) and I fear it will cause complications when used in a browser.

TLDR: I see no real advantages in having an ESM version in Node (for now).

Comment (choumx) [reply src/purifier/package.json]: Agreed, just sharing for anyone who was also curious.

Comment (fstanis) [reply src/purifier/package.json]: Ah, got it, sorry I misunderstood your previous comment.

**Comment by fstanis** [comment]
Update: it seems passing `doc.defaultView` instead of `self` is generally a bad idea: `defaultView` may be unset and this makes DOMPurify think the library is not running in a real browser, making it a singleton.
